vmm/src/vm.rs

```diff
@@ -2156,12 +2156,40 @@ impl Vm {
         Ok(table)
     }
 
+    #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+    pub fn enable_memory_log_dirty(&self) -> std::result::Result<(), MigratableError> {
+        self.vm.enable_dirty_page_tracking().map_err(|e| {
+            MigratableError::EnableDirtyLogging(anyhow!(
+                "Error enabling dirty page tracking: {}",
+                e
+            ))
+        })
+    }
+    #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+    pub fn disable_memory_log_dirty(&self) -> std::result::Result<(), MigratableError> {
+        self.vm.disable_dirty_page_tracking().map_err(|e| {
+            MigratableError::EnableDirtyLogging(anyhow!(
+                "Error disabling dirty page tracking: {}",
+                e
+            ))
+        })
+    }
+
     pub fn start_memory_dirty_log(&self) -> std::result::Result<(), MigratableError> {
+        #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+        self.enable_memory_log_dirty()?;
         self.memory_manager.lock().unwrap().start_memory_dirty_log()
     }
 
     pub fn stop_memory_dirty_log(&self) -> std::result::Result<(), MigratableError> {
-        self.memory_manager.lock().unwrap().stop_memory_dirty_log()
+        self.memory_manager
+            .lock()
+            .unwrap()
+            .stop_memory_dirty_log()?;
+
+        #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+        self.disable_memory_log_dirty()?;
+        Ok(())
     }
```

Comment:
Changes to the `vmm` crate (e.g.  to "vm.rs") should not be needed, as all MSHV specific operations should be encapsulated inside the `hypervisor` crate with its APIs "start/stop_dirty_log()" (like what the KVM backend does).

Reply:
Sorry I missed this.